integration_tests/expr_05.py

```diff
@@ -23,9 +23,12 @@ def main0():
     a = 123282374
     b = 32771
     assert test_mod(a, b) == 30643
-    a = -123282374
-    b = 32771
-    assert test_mod(a, b) == 2128
+    a = -5345
+    b = -534
+    assert a % b == -5
+    # a = -123282374
+    # b = 32771
+    # assert test_mod(a, b) == 2128
```

Comment:
This PR is almost ready to go in. We just have to make this work.
`a % b` gives `-30643` in C++ and that is the same result we get from the current implementation. Meanwhile Python gives `2128`.

Reply:
So it seems like it depends on how various programming languages implement it.

Reply:
If you have any suggestions, please let me know.

Reply:
Yes, that's right. Python uses a certain variant, they all differ for negative numbers, but agree for positive numbers. It's actually tricky to agree on one in ASR. We could add two, say Mod1, Mod2 (there might be a better name), and use one for LFortran, another one for LPython. If we add a C++ frontend, it would use one of the two.

It might be the easiest to just keep calling the manual function that we already have in LPython, for now.